collectors/python.d.plugin/postgres/postgres.chart.py

```diff
@@ -1233,3 +1257,38 @@ def create_lines(slot, lines):
     definitions[chart_name] = {
         'options': [name, title + ': ' + slot_name, units, 'replication slot files', context, chart_type],
         'lines': create_lines(slot_name, chart_template['lines'])}
+
+def add_stat_replications_chart(s,order, definitions, name, secondaries):
+    def create_lines(lines, secondaries):
+        result = list()
+        for line in lines:
+            for secondary in secondaries:
+                secondary=secondary.replace("-","")
+                metric='_'.join([secondary, line[0]])  
+                METRICS[QUERY_NAME_STAT_REPLICATION].append(metric)
+                new_line = [metric, secondary] + line[2:]
+                result.append(new_line)
+            return result
+
+    QUERY_STAT_REPLICATION[DEFAULT] = CREATE_STAT_QUERY(secondaries)
+    s.queries[query_factory(QUERY_NAME_STAT_REPLICATION, s.server_version)] = METRICS[QUERY_NAME_STAT_REPLICATION]
+    chart_template = CHARTS[name]
+    position = order.index('database_size')
+    order.insert(position, name)
+    _, title, units, _, context, chart_type = chart_template['options']
+    definitions[name] = {
+        'options': [name, title, units, 'stat replications', context, chart_type],
+        'lines': create_lines(chart_template['lines'], secondaries)}
+
+def CREATE_STAT_QUERY(secondaries):
+    STAT_QUERY = """
+    (SELECT count(1) AS on 
+    FROM pg_stat_replication 
+    WHERE application_name = '{appname}' ) as {alias}"""
```

Comment:
Can we just do a query and check if secondaries in there later in the code?

I assume it would make implementation less hacky. I see we modify global queries and local queries (s.queries) in the `add_stat_replications_chart` function.

Reply:
I've tried this, and worked...
I add a new commit.
![image](https://user-images.githubusercontent.com/76041015/110132945-9c576500-7dc3-11eb-9579-99baf777cbcb.png)
![image](https://user-images.githubusercontent.com/76041015/110132679-57333300-7dc3-11eb-83a5-5b690ab5a678.png)